crates/shared/src/price_estimation.rs

```diff
@@ -133,6 +134,24 @@ pub async fn ensure_token_supported(
     }
 }
 
+#[mockall::automock]
+#[async_trait::async_trait]
+pub trait NativePriceEstimating: Send + Sync {
```

Comment:
The return type is a plain `f64`. Could we make it clear in what unit this is? Like the comment  for price estimate

> The resulting price is how many units of sell_token needs to be sold for one unit of buy_token (sell_amount / buy_amount).

Reply:
Will do.